draft-ietf-tls-tls13.md

```diff
@@ -1782,19 +1811,6 @@ Note: In versions of TLS prior to TLS 1.3, this extension was named
 ECDSA curves. Signature algorithms are now negotiated independently (see
 {{signature-algorithms}}).
 
-Clients which offer one or more (EC)DHE cipher suites MUST send at least one
-supported NamedGroup value and servers MUST NOT negotiate any of these
-cipher suites unless a supported value was provided.
-If this extension is not provided and no alternative cipher suite is available,
-the server MUST close the connection with a fatal "missing_extension" alert.
-(see {{mti-extensions}})
-If the extension is provided, but no compatible group is offered, the
-server MUST NOT negotiate a cipher suite of the relevant type. For
-instance, if a client supplies only ECDHE groups, the server MUST NOT
-negotiate finite field Diffie-Hellman. If no acceptable group can be
-selected across all cipher suites, then the server MUST generate a
-fatal "handshake_failure" alert.
```

Comment:
It's probably worth being clear on what missing this extension means (I've seen TLS 1.2 implementation which send no curves and expect to do ECDHE, though hopefully this will be noticed by servers not tolerating it anymore). Maybe something like:

Clients which offer (EC)DHE or (EC)DHE-PSK key establishment MUST send at least one supported NamedGroup value. Servers MUST NOT negotiation an (EC)DHE group unless a supported value was provided. Note that, in that case, pure PSK key establishment is still possible.